# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/webrender_bindings/DCLayerTree.cpp
# Commit: 03f3e71928f5
# Full Hash: 03f3e71928f5db04a68e19d0af6f92d3519d50c3
# Author: agoloman <agoloman@mozilla.com>
# Date: 2025-06-12 16:31:09
# Regressor Bug: 1945683
# File Overlap Count: 1
# Description:
#   Revert "Bug 1945683 - Enable layer compositor on Windows in Nightly r=gfx-reviewers,lsalzman" for causing performance regression on Speedometer 3 on Windows 11.
#   
#   This reverts commit c7b785cbc18de37b0db02ced9285aebe7efbf580.
# ==============================================================================

diff -r b29e61ac6b83 -r 03f3e71928f5 gfx/webrender_bindings/DCLayerTree.cpp
--- a/gfx/webrender_bindings/DCLayerTree.cpp	Thu Jun 12 13:56:38 2025 +0000
+++ b/gfx/webrender_bindings/DCLayerTree.cpp	Thu Jun 12 17:05:04 2025 +0300
@@ -547,8 +547,13 @@
 }
 
 bool DCLayerTree::UseLayerCompositor() const {
+// Only allow the layer compositor in nightly builds, for now.
+#ifdef NIGHTLY_BUILD
   return UseNativeCompositor() &&
          StaticPrefs::gfx_webrender_layer_compositor_AtStartup();
+#else
+  return false;
+#endif
 }
 
 void DCLayerTree::DisableNativeCompositor() {